### PR TITLE
fix(expect): support `expect.addSnapshotSerializer`

### PR DESCRIPTION
**Pull request by eryue0220**
Part of #3964, support `expect.addSnapshotSerializer` api.

### CODE REVIEW

**Comment by eryue0220** [comment]
The error about typo checking has submitted in this pr https://github.com/denoland/std/pull/6171.

File: expect/expect.ts [hunk 1]
```diff
@@ -599,3 +600,21 @@ expect.not = {
   stringContaining: asymmetricMatchers.stringNotContaining,
   stringMatching: asymmetricMatchers.stringNotMatching,
 };
+/**
+ * `expect.addSnapshotSerializer` adds a module that formats application-specific data structures.
+ *
+ * For an individual test file, an added module precedes any modules from snapshotSerializers configuration,
+ * which precede the default snapshot serializers for built-in JavaScript types and for React elements.
```
Comment (kt3k) [inline expect/expect.ts:607]: I'm removing `for React elements` part as that phrase feels strange in this repo.

File: expect/_types.ts [hunk 2]
```diff
@@ -915,3 +915,5 @@ export interface OldSnapshotPlugin {
   ) => string;
   test: Test;
 }
+
+export type SnapshotPlugin = NewSnapshotPlugin | OldSnapshotPlugin;
```
Comment (kt3k) [inline expect/_types.ts:919]: Is `OldSnapshotPlugin` still common? How about only supporting `NewSnapshotPlugin`?

Comment (eryue0220) [reply expect/_types.ts:919]: Yes, I checked some serialize plugins, they still use `OldSnapshottPlugin` type definition, like:

[jest-serializer-vue](https://github.com/eddyerburgh/jest-serializer-vue/blob/master/index.js#L17)
[jest-serializer-html](https://github.com/algolia/jest-serializer-html/blob/master/createSerializer.js#L17)
[jest-serializer-react-helmet](https://github.com/keplersj/jest-serializer-react-helmet/blob/master/src/index.tsx#L24)
[jest-snapshot-serializer-ansi](https://github.com/ikatyang/jest-snapshot-serializer-ansi/blob/master/src/index.ts#L8)

Comment (kt3k) [reply expect/_types.ts:919]: Fair enough

**Review by kt3k — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**